Model/Product/Handler/Message/SynchronizeProductMessageHandler.php

```diff
@@ -147,6 +156,25 @@ protected function synchronizeTranslation(
         $productInformation->setShortDescription($translationValueObject->getShortDescription());
     }
 
+    protected function synchronizeProductTaxons(SynchronizeProductMessage $message, ProductInterface $product): void
+    {
+        $mainCategory = null;
+        $mainTaxonid = $message->getMainTaxonId();
+        if ($mainTaxonid) {
+            $mainCategory = $this->categoryRepository->findBySyliusId($mainTaxonid);
+        }
+        $product->setMainCategory($mainCategory);
+
+        $categories = [];
+        foreach ($message->getProductTaxons() as $productTaxonValueObject) {
+            $category = $this->categoryRepository->findBySyliusId($productTaxonValueObject->getTaxonId());
+            if ($category) {
+                $categories[] = $category;
+            }
+        }
+        $product->setProductCategories($categories);
```

Comment:
are you sure that changes will be synced? i thought we had problems with that before